XenModel/Actions/VM/VMCrossPoolMigrateAction.cs

```diff
@@ -148,13 +148,25 @@ protected override void Run()
                 if (network == null)
                     continue;
 
-                foreach (var vifRef in network.VIFs)
+                foreach (var vifRef in network.VIFs.Where(vifRef => vm.VIFs.Contains(vifRef)))
                 {
-                    if (vm.VIFs.Contains(vifRef))
-                        map.Add(new XenRef<VIF>(vifRef), new XenRef<XenAPI.Network>(pair.Value.opaque_ref));
+                    map.Add(new XenRef<VIF>(vifRef), new XenRef<XenAPI.Network>(pair.Value.opaque_ref));
                 }
             }
 
+            // Add mapping of snap VIFs
+            var snapVIFs = VM.get_snapshots(Connection.Session, vm.opaque_ref)
+                .Select(Connection.Resolve).SelectMany(snap => snap.VIFs).ToList();
+            foreach (var snapVIF in snapVIFs)
+            {
+                if (map.ContainsKey(snapVIF)) 

```

Comment:
I feel with this you can just test the negative and reduce the number of lines here. While using continue would work I don't think its really needed, if we can write less code we maintain less code.